server/src/main/java/app/friendship/FriendshipRepository.java

```diff
@@ -0,0 +1,24 @@
+package app.friendship;
+
+import app.user.User;
+import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.data.jpa.repository.Query;
+import org.springframework.stereotype.Repository;
+
+import java.util.List;
+
+@Repository
+public interface FriendshipRepository extends JpaRepository<Friendship, Long> {
+    // Finds all friendships where the specified user is the receiver and the friendship has a specific status.
+    List<Friendship> findByReceiverIdAndStatus(Integer receiverId, FriendshipStatus status);
+
+    // Finds friendships where:
+    // - The friendship has a specific status, and the specified user is the sender.
+    // OR
+    // - The friendship has a specific status, and the specified user is the receiver.
+    // This method is useful for querying friendships regardless of whether the user initiated or received the request.
+    List<Friendship> findByStatusAndSenderIdOrStatusAndReceiverId(FriendshipStatus status1, Integer senderId, FriendshipStatus status2, Integer receiverId);
+
+    // Checks if a friendship exists between two users in either direction (sender to receiver or receiver to sender).
+    boolean existsBySenderIdAndReceiverIdOrSenderIdAndReceiverId(Integer senderId, Integer receiverId, Integer receiverIdSwap, Integer senderIdSwap);

```

Comment:
very cool i didnt know that you can do this in java. But instead of making this super complicated logic you could just use reduced version 2 times. dont vorry about how many requests are sent 2 or 1 - it is not a big deal. having hard to read code is a big deal.